website/static/js/contribAdder.js

```diff
@@ -40,6 +40,25 @@ var AddContributorViewModel = oop.extend(Paginator, {
         self.parentTitle = parentTitle;
         self.async = options.async || false;
         self.callback = options.callback || function() {};
+        self.nodesOriginal = {};
+        //state of current nodes
+        self.nodesToChange = ko.observableArray();
+        self.nodesState = ko.observableArray();
```

Comment:
Shouldn't `nodesState` be an `observable` rather than an `observableArray`? The code treats it like an object rather than an Array.